packages/cli/src/services/project-checker.ts

```diff
@@ -69,3 +71,66 @@ export async function checkResourceExists(name: string, placementDir: string, ex
     })
   }
 }
+
+export async function checkCurrentDirBoosterVersion(userAgent: string): Promise<void> {
+  return checkBoosterVersion(userAgent, process.cwd())
+}
+
+async function checkBoosterVersion(userAgent: string, projectPath: string): Promise<void> {
+  const projectVersion = await getBoosterVersion(projectPath)
+  const cliVersion = userAgent.split(' ')[0].split('/')[2]
+  await compareVersionsAndDisplayMessages(cliVersion, projectVersion, projectPath)
+}
+
+async function getBoosterVersion(projectPath: string): Promise<string> {
+  const projectAbsolutePath = path.resolve(projectPath)
+  try {
+    const packageJsonContents = require(path.join(projectAbsolutePath, 'package.json'))
+    const version = packageJsonContents.dependencies['@boostercloud/framework-core']
+    const versionParts = version.replace('^','').replace('.tgz','').split('-')
+    return versionParts[versionParts.length-1]    
+  } catch (e) {
+    throw new Error(
+      `There was an error when recognizing the application. Make sure you are in the root path of a Booster project:\n${e.message}`
+    )
+  }
+}
+
+async function compareVersionsAndDisplayMessages(cliVersion: string, projectVersion: string, projectPath: string): Promise<void> {
+  if (cliVersion === projectVersion)  { return }
+  const cliVersionParts = cliVersion.split('.').map((v) => parseInt(v,10))
+  const projectVersionParts = projectVersion.split('.').map((v) => parseInt(v,10))
+  if (cliVersionParts.length !== projectVersionParts.length) {
+    throw new Error(`Versions must have the same length. CLI version: ${cliVersion}. Project Booster version: ${projectVersion}`)
+  }
+  if (projectVersionParts.length !== 3) {
+    throw new Error(`Versions must follow semantic convention X.Y.Z | CLI version: ${cliVersion}. Project Booster version: ${projectVersion}`)
+  }
+  if (cliVersionParts[0] === projectVersionParts[0]) {
+    if (cliVersionParts[1] === projectVersionParts[1]) {
+      //differences in the 'fix' part
+      if (cliVersionParts[2] !== projectVersionParts[2]) {
+        logger.info(`WARNING: Project Booster version differs in the 'fix' section. CLI version: ${cliVersion}. Project Booster version: ${projectVersion}`)
+      }
```

Comment:
Instead of nesting `if`s, I'd create a check function for it, and appropriate action functions as well:

```suggestion
type Semver = [number, number, number]

function semverDifferentFixVersion(a: Semver, b: Semver): bool {
  return a[0] === b[0] && a[1] === b[1] && a[2] !== b[2]
}

function warnDifferentFixVersion(cliVersion: string, projectVersion: string){
        logger.info(`WARNING: Project Booster version differs in the 'fix' section. CLI version: ${cliVersion}. Project Booster version: ${projectVersion}`)
}

// now in the code...
if(semverDifferentFixVersion(cliVersionParts, projectVersionParts) {
  warnDifferentFixVersion(cliVersion, projectVersion)
}
```

Reply:
A better approach is to encapsulate this logic in Semver class